components/WeatherForecast/index.js

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+const WeatherContainer = styled.div`
+  margin-top: 20px;
+  background-color: #f5f8fb;
+  border-radius: 8px;
+  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+`;
+
+const WeatherTitle = styled.h2`
+  color: #3f72af;
+  font-size: 18px;
+  margin: 20px 0;
+  text-align: center;
+`;
+
+const WeatherInfoList = styled.ul`
+  list-style: none;
+  padding: 0;
+  margin: 0;
+`;
+
+const WeatherInfoItem = styled.li`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: 10px;
+  border-bottom: 1px solid #ddd;
+  background-color: #f9f9f9;
+
+  &:last-child {
+    border-bottom: none;
+  }
+`;
+
+const WeatherInfoGroup = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  flex: 1;
+`;
+
+const WeatherInfoTitle = styled.p`
+  font-weight: bold;
+  margin-bottom: 5px;
+`;
+
+const WeatherInfoValue = styled.p`
+  font-weight: normal;
+`;
+
+const WeatherForecast = ({ location, startDate, endDate }) => {
+  const [weatherData, setWeatherData] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchWeatherForecast = async () => {
+      try {
+        const response = await fetch(
+          `https://api.weatherbit.io/v2.0/forecast/daily?key=${process.env.NEXT_PUBLIC_WEATHER_API}&city=${location}&start_date=${startDate}&end_date=${endDate}`
```

Comment:
❗ Calling a 3rd party API directly from the client is against `Next.js` architecture because you are exposing your API key (and by the way the server responds with a 403 forbidden)! 
So please refactor to an internal API route which serves as a kind of middleware/passthrough and performs your request to this weatherbit API.
Following this approach, you can get rid of your effect hook and implement `useSWR` to query your internal API.